app/models/task.rb

```diff
@@ -36,4 +36,8 @@ def archived?
   def days_left
     (self.due_date - Date.today).to_i if self.due_date
   end
+
+  def overdue?
+    true unless days_left.nil? || days_left >= 0 || archived?
```

Comment:
Oh yeah, totally!  This is great.  That's a much more straightforward way to express things. 👍